### PR TITLE
Initial update of dependencies to Jakarta EE 10 versions.

### PR DESCRIPTION
**Pull request by arjantijms**
First batch

### CODE REVIEW

**Comment by arjantijms** [comment]
Blocker: https://github.com/eclipse-ee4j/metro-wsit/issues/138

**Comment by arjantijms** [comment]
Blocker: https://hibernate.atlassian.net/browse/HV-1857

**Comment by arjantijms** [comment]
Blocker: https://github.com/eclipse-ee4j/eclipselink/issues/1371

**Comment by arjantijms** [comment]
Blocker https://github.com/WASdev/standards.jsr352.jbatch/issues/74

**Comment by arjantijms** [comment]
Blocker https://github.com/eclipse-ee4j/tyrus/issues/762

File: nucleus/parent/pom.xml [hunk 1]
```diff
@@ -83,41 +83,41 @@
...
-        <jakarta.jaxb-impl.version>3.0.2</jakarta.jaxb-impl.version>
+        <jakarta.jaxb-api.version>4.0.0-RC2</jakarta.jaxb-api.version>
+        <jakarta.jaxb-impl.version>4.0.0-M1</jakarta.jaxb-impl.version>
 
         <!-- Jakarta REST -->
         <jax-rs-api.spec.version>3.0</jax-rs-api.spec.version>
-        <jax-rs-api.impl.version>3.0.0</jax-rs-api.impl.version>
-        <jersey.version>3.0.3</jersey.version>
+        <jax-rs-api.impl.version>3.1.0</jax-rs-api.impl.version>
+        <jersey.version>3.1.0-SNAPSHOT</jersey.version>
```
Comment (jansupol) [inline nucleus/parent/pom.xml:109]: Jersey has 3.1.0-M1 in sonatype already

Comment (arjantijms) [reply nucleus/parent/pom.xml:109]: Thanks, I'll try that one

Comment (arjantijms) [reply nucleus/parent/pom.xml:109]: Unfortunately M1 is not strong enough, I get the following error again, which blocks the startup of the server completely:

```
3. java.lang.IllegalStateException: Unable to resolve
    org.glassfish.main.web.jersey-mvc-connector [31]
    missing requirement
        &(package = org.glassfish.jersey.server.mvc.jsp) (version >= 3.1.0) (!(version >= 4.0.0))
        caused by:
            Unable to resolve
                org.glassfish.jersey.ext.jersey-mvc-jsp [158]
                missing requirement
                    &(package = org.glassfish.jersey.server.mvc.internal) (version >= 3.1.0) (!(version >= 4.0.0))
                    caused by:
                        Unable to resolve
                            org.glassfish.jersey.ext.jersey-mvc [276]
                            missing requirement
                                &(package = jakarta.servlet) (version >= 5.0.0) (!(version >= 6.0.0)))]]
```

Comment (arjantijms) [reply nucleus/parent/pom.xml:109]: I can try to hotpatch M1 though and try to make it as good as the current 3.1 branch.

Comment (jansupol) [reply nucleus/parent/pom.xml:109]: Right, 3.1.0-M1 does not have the servlet OSGi headers set properly, yet

Comment (arjantijms) [reply nucleus/parent/pom.xml:109]: M1 is probably also missing the CDI fixes. If I hot patch the archive, I get this:

```
 java.lang.NoSuchMethodError: 'jakarta.enterprise.inject.spi.InjectionTarget jakarta.enterprise.inject.spi.BeanManager.createInjectionTarget(jakarta.enterprise.inject.spi.AnnotatedType)'
	at org.glassfish.jersey.ext.cdi1x.servlet.internal.CdiExternalRequestScopeExtension.afterBeanDiscovery(CdiExternalRequestScopeExtension.java:72)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:77)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:568)
	at org.jboss.weld.injection.StaticMethodInjectionPoint.invoke(StaticMethodInjectionPoint.java:95)
```

It's probably better for GF to keep using the branch for now, and wait till there's a M2 with the necessary fixes.

**Comment by arjantijms** [comment]
Blocker https://github.com/eclipse-ee4j/metro-wsit/issues/148

**Comment by arjantijms** [comment]
Blocker: https://github.com/eclipse-ee4j/eclipselink/issues/1380

File: appserver/tests/quicklook/QuickLook_Test_Instructions.html [hunk 2]
```diff
@@ -237,8 +237,8 @@ <H1 STYLE="margin-left: 0.83in"><FONT FACE="Trebuchet MS, sans-serif"><FONT SIZE
 deploy runtest&quot;</P>
 <P STYLE="margin-bottom: 0in">Since QL runs in multiple profiles, the
 test cases need to be defined in the 3 test lists (under
-quicklook/testng dir) for the profiles: testng_wd.xml for the Web
-profile, testng_gd.xml for Glassfish profile, testng_debug.xml for
+quicklook/testng dir) for the profiles: testng_web_profile.xml for the Web
+profile, ttestng_full_profile.xml for Glassfish full profile, testng_debug.xml for
```
Comment (pzygielo) [inline appserver/tests/quicklook/QuickLook_Test_Instructions.html:241]: extra `t` in `ttestng_full_profile.xml`?

Comment (arjantijms) [reply appserver/tests/quicklook/QuickLook_Test_Instructions.html:241]: > ttestng_full_profile.xml

That's an amazing eagle eyed spot!

File: nucleus/parent/pom.xml (hunk 1 above)
Comment (arjantijms) [reply nucleus/parent/pom.xml]: -m2 now available, and added. Thanks!

**Review by jGauravGupta — APPROVED** [review]
LGTM

**Pull request merged**